vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -419,7 +419,7 @@ def __init__(
         # Use expert parallelism instead of tensor parallelism?
         vllm_config = get_current_vllm_config()
         use_ep = (vllm_config.parallel_config.enable_expert_parallel
-                  and self.tp_size > 1)
+                  and self.tp_size * self.dp_size > 1)
```

Comment:
This fix is good. Without it, everything will work OK but the MoE layers will run with TP instead of EP (pretty sure that's true at least). Probably worth putting into its own PR